embrace-android-sdk/src/main/cpp/utils/string_utils.c

```diff
@@ -1,14 +1,20 @@
 #include "string_utils.h"
 #include "../schema/stack_frames.h"
 
-void emb_strncpy(char *dst, const char *src, size_t len) {
-    if (dst == NULL || src == NULL) {
+/**
+ * Copy a string from source to destination. If the source is larger than the destination, it will be truncated.
+ * @param destination The destination buffer.
+ * @param source The source buffer.
+ * @param destination_len The length of the destination buffer.
+ */
+void emb_strncpy(char *destination, const char *source, size_t destination_len) {
+    if (destination == NULL || source == NULL) {
         return;
     }
     int i = 0;
-    while (i <= len) {
-        char current = src[i];
-        dst[i] = current;
+    while (i < destination_len && source[i] != '\0') {
```

Comment:
This change means the null termination character is not copied to the destination array, which could cause various problems downstream as lots of C code assumes that strings must have that character.